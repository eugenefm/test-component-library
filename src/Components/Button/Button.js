import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { variant, color, space, typography } from 'styled-system';

const StyledButton = styled('button')(
	{
		textTransform: 'uppercase',
		letterSpacing: '1px'
	},
	variant({
		variants: {
			primary: {
				color: 'white',
				bg: 'blue',
				padding: 2,
				border: 'none'
			},
			secondary: {
				color: 'blue',
				bg: 'white',
				padding: 2,
				border: '1px solid blue'
			}
		}
	}),
	color,
	space,
	typography
);

const Button = (props) => {
	const { children, ariaLabel, onClick, variant } = props;
	return (
		<StyledButton
			variant={variant}
			fontSize={0}
			onClick={onClick}
			aria-label={ariaLabel}
		>
			{children}
		</StyledButton>
	);
};

Button.propTypes = {
	children: PropTypes.node.isRequired,
	ariaLabel: PropTypes.string,
	onClick: PropTypes.func.isRequired,
	variant: PropTypes.string
};

Button.defaultProps = {
	variant: 'primary'
};

export default Button;
