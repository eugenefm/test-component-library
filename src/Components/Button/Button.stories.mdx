import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, select} from '@storybook/addon-knobs'
import Button from './Button';

<Meta title='Button' component={Button} />

# Button
This is the button that I made, isn't it awesome.

## Properties
Here are all the props that it takes, which are automatically defined using `propTypes` and `defaultProps`.

<Props of={Button} />

## Basic usage
Indendation on the code preview looks a little broken in the current version of Storybook - [see this issue](https://github.com/storybookjs/storybook/issues/8078).

<Preview>
	<Story name='Primary Button'>
		<Button 
			aria-label='Call to Action' 
			onClick={action('Button Click')}
		>
			Call to Action
		</Button>
	</Story>
	<Story name='Secondary Button'>
		<Button 
			variant='secondary' 
			aria-label='Call to Action' 
			onClick={action('Button Click')}
		>
			Call to Action
		</Button>
	</Story>
</Preview>

This additional story will be hidden from the docs in `Storyboook 5.3.0`
<Story name='Button with Knobs' parameters={{ docs: { disable: true }}}  decorators={[withKnobs]}>
	<Button 
		variant={select('Variant', {
			primary: 'primary',
			secondary: 'secondary'
		}, 'primary')} 
		aria-label='Call to Action' 
		onClick={action('Button Click')}
	>
		{text('Text','Call to Action')}
	</Button>
</Story>
