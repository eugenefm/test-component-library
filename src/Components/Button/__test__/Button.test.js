import React from 'react';
import ReactDom from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import Button from '../Button';

afterEach(cleanup);

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDom.render(
		<Button onClick={() => console.log('Hello')}>Button</Button>,
		div
	);
	ReactDom.unmountComponentAtNode(div);
});

it('render Button Component correctly', () => {
	const { getByText } = render(
		<Button onClick={() => console.log('Hello')}>Hello World</Button>
	);
	expect(getByText('Hello World')).toExist;
});

it('matches snapshot', () => {
	const tree = renderer
		.create(
			<Button onClick={() => console.log('Hello')}>Call to action</Button>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
