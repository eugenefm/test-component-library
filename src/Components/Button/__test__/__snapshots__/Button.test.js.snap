// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="sc-bdVaJa gakZlE"
  fontSize={0}
  onClick={[Function]}
>
  Call to action
</button>
`;
